`gravatar-ui` - Empty state in UI components

hamorillo opened the pull request:
Closes #132 

### Description

This PR adds the `empty state` in all the UI components.

| Light | Dark |
|----------|-----------|
|![image](https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/6eb97a90-5d6d-431c-bc34-b6e621330db3)|![image](https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/afb143b8-1ea7-4a24-bb5a-e39e97e54e5e)|

### Testing Steps

Using the demo app, in the `Profile` tab, look for an email without a Gravatar account, (ex: `gravatarr@automattic.com`) and verify that the empty views look as expected. 

While reviewing the views switch between `light` / `dark` and `system` theme.

Verify that tapping over the `Gravatar` icon brings you to the `www.gravatar.com`.
Verify that tapping over the `Claim profile` or any other part of the profile components brings you to the `Sign-up` page.

Smoke test that loading and loaded states also looks as expected.

hamorillo committed:
Empty state for Profile and the components it depends on

hamorillo committed:
Empty state for ProfileSummary and the components it depends on

hamorillo committed:
Empty state for LargeProfile and the components it depends on

hamorillo committed:
Empty state for LargeProfileSummary and the components it depends on

hamorillo commented on gravatar-ui/src/main/java/com/gravatar/ui/components/atomic/Avatar.kt:
```diff
@@ -77,6 +88,16 @@ public fun Avatar(
         }
+
+        UserProfileState.Empty -> Avatar(
+            model = if (isNightModeEnabled()) {
+                R.drawable.empty_profile_avatar_dark
+            } else {
+                R.drawable.empty_profile_avatar
+            },
```
At first I tried using the folder `drawable-night` and it works well when you switch between light and dark modo in the device. However, we you override the `uiMode` as [we do in the demo app](https://github.com/Automattic/Gravatar-SDK-Android/blob/d43fab05636329659b124b49f525b739200f415c/demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt#L172), it still shows the drawable for the device theme. 

For that reason I've written this workaround.

maxme commented on gravatar-ui/src/main/java/com/gravatar/ui/components/atomic/ViewProfileButton.kt:
```diff
@@ -50,11 +51,28 @@ public fun ViewProfileButton(
+}
+
+@Composable
+private fun ViewProfileButton(
```
I think we can rename that to something more generic, since we can change the text and the url, this button can do other actions.

That reminded me, we should create an "EditProfileButton" component as well (probably needed for Pocket Casts integration).

hamorillo replied:
When using the SDK, you can modify the text but not the URL. What about `ProfileButton` which, can include `ClaimProfile`, `EditProfile` and `NavigateToProfile`. I guess we can iterate over this.

maxme reviewed (approved):
LGTM :shipit: 

One comment though:

> Verify that tapping over the `Gravatar` icon and `Claim profile` brings you to the `www.gravatar.com`.

This works fine, but from the design document, I see:

> Clicking on the Gravatar logo would take the user to Gravatar.com.
> Clicking anywhere else on the Profile Card will take the user directly to the Sign-up flow.

I'm not sure if this was discussed elsewhere and updated. If we have to, we can create another ticket for this.

hamorillo committed:
Open Profile URL when clicking over ViewProfileButton

hamorillo committed:
Add click over empty profile component

The Sign-up page should be opened when an empty profile component is clicked.

hamorillo commented:
> This works fine, but from the design document, I see:
> 
> > Clicking on the Gravatar logo would take the user to Gravatar.com.
> > Clicking anywhere else on the Profile Card will take the user directly to the Sign-up flow.
> 
> I'm not sure if this was discussed elsewhere and updated. If we have to, we can create another ticket for this.

I've updated the behavior when clicking the empty profile components. I'm not completely sure If I like that empty / loaded states have different behaviors, but I guess we can discuss and iterate over it.

@maxme Could you give another review round, please? 🙏🏼

maxme reviewed (approved):
LGTM :shipit:
